pytorch_lightning/lite/wrappers.py

```diff
@@ -131,6 +131,7 @@ def __iter__(self) -> Union[Iterator[Any], Generator[Any, None, None]]:
         iterator = iter(self._dataloader)
         if self._device is None:
             yield from iterator
+            return None
```

Comment:
was having no return a bug? or not because the iterator would've been exhausted in the loop below?

Reply:
Not a bug: 
The iterator is exhausted but mypy does not know that. It pretends that self.device can be None after the if statement.

Reply:
```suggestion
            return
```

None is not required here

Reply:
this is mypy business, always have to be explicit

Reply:
That's not true. I ran it locally and the everything is green.

Mypy only requires it when there's another return

See how only `fn2` has an error here: https://mypy-play.net/?mypy=latest&python=3.10&gist=6bd880ade68f34c5ff86f47662022302